roles/keepalived/tasks/main.yml

```diff
@@ -42,15 +43,29 @@
   notify: "restart keepalived"
   tags: keepalived_conf, keepalived
 
+- name: Parse keepalived_instances as YAML
+  ansible.builtin.set_fact:
+    keepalived_instances_parsed: "{{ keepalived_instances | from_yaml }}"
+  when: add_balancer is not defined or not add_balancer|bool
+  tags: keepalived_conf, keepalived
+
 - name: Generate conf file "/etc/keepalived/keepalived.conf"
   ansible.builtin.template:
     src: templates/keepalived.conf.j2
     dest: /etc/keepalived/keepalived.conf
+    owner: root
+    group: root
+    mode: "0644"
   notify: "restart keepalived"
+  vars:
+    keepalived_instances: "{{ keepalived_instances_parsed }}"
   when: add_balancer is not defined or not add_balancer|bool
   tags: keepalived_conf, keepalived
 
-- block:  # for add_balancer.yml
+- name: Generate conf file "/etc/keepalived/keepalived.conf" for add_balancer
+  when: add_balancer is defined and add_balancer|bool
+  tags: keepalived_conf, keepalived
```

Comment:
please stick to one code format. Move the conditions and tags down in the block.

Reply:
Done :)